app/models/package_manager/puppet.rb

```diff
@@ -0,0 +1,75 @@
+module PackageManager
+  class Puppet < Base
+    HAS_VERSIONS = true
+    HAS_DEPENDENCIES = true
+    BIBLIOTHECARY_SUPPORT = false
+    BIBLIOTHECARY_PLANNED = true
+    URL = 'https://forge.puppet.com'
+    COLOR = '#302B6D'
+
+    def self.project_names
+      offset = 0
+      projects = []
+      while true
+        results = get_json("https://forgeapi.puppetlabs.com/v3/modules?limit=100&offset=#{offset}")['results'].map { |result| result['slug'] }
+	break if results == []
+	projects += results
+	offset +=100
+      end
+      projects
+    end
+
+    def self.project(name)
+      get_json("https://forgeapi.puppetlabs.com/v3/modules/#{name}")
+    end
+
+    def self.mapping(project)
+      current_release = project['current_release']
+      metadata = current_release['metadata']
+      {
+        name: project['slug'],
+        repository_url: metadata['source'],
+        description: metadata['description'],
+        keywords_array: current_release['tags'],
+        licenses: metadata['license']
+      }
+    end
+
+    def self.versions(project)
+      project['releases'].map do |release|
+        {
+          number: release['version'],
+          published_at: release['created_at']
+        }
+      end
+    end
+
+    def self.dependencies(name, version, project)
+      metadata = project['current_release']['metadata']
```

Comment:
This should be the dependencies for the given version, looks like it's only setting dependencies for the current version on every version here

Reply:
Unfortunately the API doesn't expose this information as part of getModule: https://forgeapi.puppetlabs.com/#!/module/getModule

I've opened an issue against the Forge API track whether we can expose that information.
https://tickets.puppetlabs.com/browse/FORGE-394

Reply:
Looks like we can use the release endpoint instead for that: https://forgeapi.puppetlabs.com/v3/releases/puppetlabs-ntp-6.1.0

Reply:
Ah ha. Excellent. I'd missed that. Will look to update to use that endpoint.

Reply:
I've pushed a new commit which uses this endpoint. dependencies should now be per version.